Gems/Streamer/StreamerProfiler/Code/Source/StreamerProfilerSystemComponent.cpp

```diff
@@ -0,0 +1,776 @@
+                values.AddValue(azlossy_cast<float>(value.m_value));
+            }
+            else if constexpr (AZStd::is_same_v<Type, AZ::IO::Statistic::Time> || AZStd::is_same_v<Type, AZ::IO::Statistic::TimeRange>)
+            {
+                values.AddValue(azlossy_cast<float>(value.m_value.count()));
+            }
+        };
+        AZStd::visit(visitor, value);
+
+        auto valueRetriever = [](void* data, int idx) -> float
+        {
+            return (*reinterpret_cast<GraphStore*>(data))[idx];
+        };
+
+        if (useHistogram)
+        {
+            ImGui::PlotHistogram("", valueRetriever, &values, GraphStoreElementCount, 0, nullptr, values.GetMin(), values.GetMax());
+        }
+        else
+        {
+            ImGui::PlotLines("", valueRetriever, &values, GraphStoreElementCount, 0, nullptr, values.GetMin(), values.GetMax());
+        }
+#endif // #if defined(IMGUI_ENABLED)
+    }
+
+    void StreamerProfilerSystemComponent::DrawStatisticValue(const AZ::IO::Statistic::Value& value, float capturedMin, float capturedMax)
+    {
+#if defined(IMGUI_ENABLED)
+        auto visitor = [capturedMin, capturedMax](auto&& value)
+        {
+            using Type = AZStd::decay_t<decltype(value)>;
+            if constexpr (AZStd::is_same_v<Type, bool>)
+            {
+                static ImColor colorEnabled = ImColor(0.0f, 1.0f, 0.0f, 0.75f);
+                static ImColor colorDisabled = ImColor(1.0f, 0.0f, 0.0f, 0.75f);
+                ImGui::TableSetBgColor(ImGuiTableBgTarget_CellBg, value ? colorEnabled : colorDisabled);
+                ImGui::Text(value ? "True" : "False");
+            }
+            else if constexpr (AZStd::is_same_v<Type, double>)
+            {
+                ImGui::Text("%.2f", value);
+                DrawToolTipV("Min: %.2f\nMax: %.2f", capturedMin, capturedMax);
+            }
+            else if constexpr (AZStd::is_same_v<Type, AZ::IO::Statistic::FloatRange>)
+            {
+                if (value.m_min != AZStd::numeric_limits<decltype(value.m_min)>::max() &&
+                    value.m_max != AZStd::numeric_limits<decltype(value.m_max)>::min())
+                {
+                    ImGui::Text("%.2f", value.m_value);
+                    DrawToolTipV("Min: %.2f\nMax: %.2f", value.m_min, value.m_max);
+                }
+                else
+                {
+                    ImGui::Text("Unused");
+                }
+            }
+            else if constexpr (AZStd::is_same_v<Type, AZ::s64>)
+            {
+                ImGui::Text("%lld", value);
+                DrawToolTipV("Min: %i\nMax: %i", azlossy_cast<int>(capturedMin), azlossy_cast<int>(capturedMax));
+            }
+            else if constexpr (AZStd::is_same_v<Type, AZ::IO::Statistic::IntegerRange>)
+            {
+                if (value.m_min != AZStd::numeric_limits<decltype(value.m_min)>::max() &&
+                    value.m_max != AZStd::numeric_limits<decltype(value.m_max)>::min())
```

Comment:
Instead of using numeric limits to determine if a value is set or not, can we use [NAN](https://www.cplusplus.com/reference/cmath/NAN/) instead?

Reply:
That would make the code overall more complex because currently adding a value to a ranged metric would pick the smallest or largest of the new value and the stored values. If NAN is used an extra set of checks would need to be done for NAN first. Also the code for the `FloatRange` would diverge from the `IntegerRange` and I'd like to keep the symmetry.